addons/website/models/website_visitor.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+from odoo.http import request
+from datetime import datetime, timedelta
+from hashlib import md5
+
+
+class WebsiteVisitor(models.Model):
+    _name = 'website.visitor'
+    _description = 'Website Visitor'
+    _order = 'last_connection_date DESC'
+
+    name = fields.Char('Name', default='Website Visitor')
+    website_id = fields.Many2one('website', "Website", readonly=True)
+    partner_ids = fields.Many2many('res.partner')
+    partner_count = fields.Integer('# Partners', compute="_compute_partner_count")
+    create_date = fields.Datetime('First connexion date', readonly=True)
+    last_connection_date = fields.Datetime('Last Connection', help="Last page view date", index=True, readonly=True)
+    country_id = fields.Many2one('res.country', 'Country', readonly=True)
+    country_flag = fields.Binary(related="country_id.image", string="Country Flag")
+    lang_id = fields.Many2one('res.lang', string='Language', help="Language from the website when visitor has been created")
+    visit_count = fields.Integer('Number of visits', default=1, readonly=True, help="A new visit is considered if last connection was more than 8 hours ago.")
+    page_ids = fields.One2many('website.visitor.page', 'visitor_id', string='Visited Pages History')
```

Comment:
I would avoid naming page_ids a o2m to visitor.page when having a m2m to page that is called differently ;) .

We could have: visitor_page_ids, visitor_page_count, page_ids.